lib/ofx/parser/datetime.ex

```diff
@@ -1,67 +1,127 @@
 defmodule Ofx.Parser.Datetime do
   @moduledoc false
 
-  import String, only: [to_integer: 1]
-
   alias Ofx.Parser.Error
 
-  @regex ~r'^(?<y>\d{4})(?<M>\d{2})(?<d>\d{2})(?<h>\d{2})?(?<m>\d{2})?(?<s>\d{2})?(\.\d{3})?.*?(\[(?<signal>[-+]?)(?<th>\d{1,2})\.?(?<tm>\d{1,2})?)?'
-
-  @seconds_in_an_hour 3600
-  @seconds_in_a_minute 60
+  defstruct [:input, :datetime, :date, :time, :offset, :timezone, :zone_abbreviation]
 
-  def format(date_string) do
-    @regex
-    |> Regex.named_captures(date_string)
-    |> build_naive_date_time()
-    |> shift_tz_to_uct0()
+  def format(input) do
+    %__MODULE__{input: input}
+    |> extract_timezone
```

Comment:
```suggestion
    |> extract_timezone()
```
:nail_care: Could you add the parenthesis to all function calls that it is missing?